packages/roosterjs-editor-dom/lib/table/VTable.ts

```diff
@@ -239,23 +240,51 @@ export default class VTable {
                 break;
 
             case TableOperation.DeleteRow:
-                this.forEachCellOfCurrentRow((cell, i) => {
-                    let nextCell = this.getCell(this.row + 1, i);
+                const deleteRowHandler = (cell: VCell, i: number, rowIndex: number) => {
+                    let nextCell = this.getCell(rowIndex + 1, i);
                     if (cell.td && cell.td.rowSpan > 1 && nextCell.spanAbove) {
                         nextCell.td = cell.td;
                     }
-                });
-                this.cells.splice(this.row, 1);
+                };
+                if (this.selection) {
+                    const { firstCell, lastCell } = this.selection;
+                    for (let rowIndex = firstCell.y; rowIndex <= lastCell.y; rowIndex++) {
+                        this.forEachCellOfRow(rowIndex, (cell: VCell, i: number) => {
+                            deleteRowHandler(cell, i, rowIndex);
+                        });
+                    }
+                    this.cells.splice(firstCell.y, lastCell.y - firstCell.y + 1);
+                } else {
+                    this.forEachCellOfCurrentRow((cell, i) => {
+                        deleteRowHandler(cell, i, this.row);
+                    });
+                    this.cells.splice(this.row, 1);
+                }

```

Comment:
So I'd like to see these if.. and else.. block merged together.

The difference of them are
1. Delete single row vs delete multiple row
2. Read current row from this.row vs from this.selection

You can read the row numbers first, then always assume delete multiple rows. When delete single, it can be treated the first row and last row are the same. So something like:
```ts
const firstRow = this.selection ? this.selection.firstCell.y : this.row;
const lastRow = this.selection ? this.selection.lastCell.y : this. row;
for (from firstRow to last row) ...
```

Reply:
Then no need to have the function "deleteRowHandler". It can be just inline into the loop

Reply:
Then no need to have the function "deleteRowHandler". It can be just inline into the loop